File: 38-CountAndSay/38-CountAndSay.java
// Last updated: 7/24/2025, 8:35:54 AM
class Solution {
    public String countAndSay(int n) {
        if(n==1)
            return "1";
        StringBuffer s = new StringBuffer();
        s.append('1');
        for(int i=2;i<=n;i++){
            StringBuffer next = new StringBuffer();
            int f=1,k=0;
            char c=s.charAt(0);
            for(int j=1;j<s.length();j++){
                if(c==s.charAt(j)){
                    f++;
                }
                else{
                    next.append(f).append(c);
                    f=1;
                    c=s.charAt(j);
                }
            }
            next.append(f).append(c);
            s = next;
        }
        return s.toString();
    }
        
}

File: 38-CountAndSay/38-CountAndSay.cpp
// Last updated: 7/24/2025, 8:36:00 AM
class Solution {
public:
    string countAndSay(int n) {
        vector<string> dp(n);
        dp[0] = "1";
        for (int i = 1; i < n; i++) {
            // dp[i]
            string s = dp[i - 1];
            s += '/';
            int count = 1;
            string ans = "";
            for (int j = 0; j < s.length() - 1; j++) {
                if (s[j] == s[j + 1])
                    count++;
                else {
                    ans += count + '0';
                    ans += s[j];
                    count = 1;
                }
            }
            dp[i] = ans;
        }
        return dp[n - 1];
    }
};